src/core/store/modules/app/app.actions.ts

```diff
@@ -34,11 +36,16 @@ const clearUserAppData = createAsyncThunk<void, void, ThunkAPI>('app/clearUserAp
 /*                                 Fetch Data                                 */
 /* -------------------------------------------------------------------------- */
 
-const initApp = createAsyncThunk<void, void, ThunkAPI>('app/initApp', async (_arg, { dispatch, getState }) => {
+const initApp = createAsyncThunk<void, void, ThunkAPI>('app/initApp', async (_arg, { dispatch, getState, extra }) => {
+  const { CONTRACT_ADDRESSES } = extra.config;
   const { wallet, network } = getState();
-  if (inIframe()) {
+  if (inLedgerIframe()) {
+    if (network.current !== 'mainnet') await dispatch(NetworkActions.changeNetwork({ network: 'mainnet' }));
     await dispatch(WalletActions.walletSelect({ walletName: 'Iframe', network: 'mainnet' }));
-  } else if (wallet.name) {
+    await dispatch(PartnerActions.changePartner({ partnerId: 'ledger', address: CONTRACT_ADDRESSES.LEDGER }));
```

Comment:
This logic could even live in a `handleInLedgerIFrame()`

Reply:
I would agree to move part of the code to another thunk if we have a need to reuse the function. For now it seems fine to me, since it would only be called here.